### PR TITLE
Add FRS configuration to e2e test pipeline

### PR DESCRIPTION
**Pull request by curtisman**
- Add configuration to target FRS service endpoint for our e2e tests
- Add shorten name for r11s and t9s for the driver type name and also `npm run test:realsvc:*` scripts.
- Refactor the code that parse the env for the endpoint info for the r11s test driver and introduce a JSON format
- Add option to specify different service endpoint for the tests that use the r11s driver.

### CODE REVIEW

**Review by skylerjokiel — APPROVED** [review]
Thanks! 🍁

File: tools/pipelines/test-real-service.yml [hunk 1]
```diff
@@ -80,16 +80,35 @@ stages:
...
+            fluid__test__driver__r11s: $(automation-fluid-test-driver-r11s)
             FLUID_TEST_LOGGER_PKG_PATH: ${{ variables.testWorkspace }}/node_modules/@ff-internal/aria-logger # Contains getTestLogger impl to inject
           inputs:
             command: 'custom'
             workingDir: ${{ variables.testWorkspace }}/node_modules/@fluidframework/test-end-to-end-tests
             customCommand: 'run test:realsvc:routerlicious:report'
 
+  # end-to-end tests routerlicious
+  - stage:
+    displayName: e2e - routerlicious
```
Comment (heliocliu) [inline tools/pipelines/test-real-service.yml:92]:
```suggestion
    displayName: e2e - frs
```
or anything else to differentiate

Comment (heliocliu) [inline tools/pipelines/test-real-service.yml]:
```suggestion
  # end-to-end tests frs
  - stage:
    displayName: e2e - frs
```
or anything else to differentiate

**Review by heliocliu — APPROVED** [review]

File: tools/pipelines/test-real-service.yml (hunk 1 above)
Comment (curtisman) [reply tools/pipelines/test-real-service.yml]: fixed.

**Pull request merged**